Fix build on 32-bit

dpino opened the pull request:
#### f74166b1ad83a65cf64fd8523054de9b351c5dae
<pre>
Fix build on 32-bit

Reviewed by Yusuke Suzuki and Justin Michaud.

I attempted to build WebKitGTK on Debian 12 32-bit using only system
libraries (&apos;--no-experimetal-features&apos;).  Exactly the same build
procedure as the WebKitGTK Debian 12 64-bit bot, but on 32-bit.

I got a couple of build errors, which this commit fixes.

* Source/JavaScriptCore/CMakeLists.txt:
* Source/WebCore/dom/EventTarget.h:

Canonical link: <a href="https://commits.webkit.org/300313@main">https://commits.webkit.org/300313@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/22ab3956d7b35eace7dbf769af72b5331658b488

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122070 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41772 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32442 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/128634 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/74164 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/123946 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42487 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50366 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92784 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61667 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125022 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33884 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109316 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73440 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32897 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/27482 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/72131 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/114218 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/103391 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27673 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/131394 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/120596 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/49009 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37275 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101345 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49383 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105530 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101216 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46585 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24700 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45746 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19320 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48866 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/54600 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/150754 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/48336 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/38574 "Found 1 new JSC stress test failure: stress/dont-link-virtual-calls-on-compiler-thread.js.no-llint (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51686 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/50016 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Constellation commented on Source/WebCore/dom/EventTarget.h:
```diff
@@ -47,6 +46,11 @@
+#if !USE(SYSTEM_MALLOC)
+#include <bmalloc/TZoneHeap.h>
+#include <bmalloc/bmalloc.h>
+#endif
```
This does not look right. Do we need to include bmalloc headers?

dpino replied:
Right, 'bmalloc.h' was not adtually needed.

justinmichaud replied:
I think this means that we should include wtf/TZoneMalloc instead

Constellation replied:
Why isn't it wtf/TZoneMalloc.h? Also, SYSTEM_MALLOC guard is not necessary

dpino replied:
Header `wtf/TZoneMalloc.h` is already included in the file. It doesn't seem header `<bmalloc/TZoneHeap.h>` is necessary as it's already included via `wtf/TZoneMalloc.h`, so I removed it.

dpino replied:
FWIW, the inclusion of header `bmalloc/TZoneHeap.h` was causing a build error in WebKitGTK 32-bit (Debian 12):

```
/home/buildbot/worker/webkit/WebKitBuild/GTK/Release/WebCore/PrivateHeaders/WebCore/EventTarget.h:50:10: fatal error: bmalloc/TZoneHeap.h: No such file or directory
   50 | #include <bmalloc/TZoneHeap.h>
	  |          ^~~~~~~~~~~~~~~~~~~~~
```

To fix this build error I followed the same pattern as other files that also include `<bmalloc/TZoneHeap.h>`, such as https://github.com/WebKit/WebKit/blob/main/Source/WebKit/Platform/IPC/Connection.h#L87 and https://github.com/WebKit/WebKit/blob/main/Source/WebKit/Platform/IPC/Decoder.h#L48.

Constellation reviewed (changes requested):
Please include wtf headers

Constellation reviewed (changes requested):
See Justin's comment. Including bmalloc header is not correct

Constellation reviewed (changes requested):
Remove NO TEST comment in the commit log.

dpino committed:
Fix build on 32-bit

Reviewed by Yusuke Suzuki and Justin Michaud.

I attempted to build WebKitGTK on Debian 12 32-bit using only system
libraries ('--no-experimetal-features').  Exactly the same build
procedure as the WebKitGTK Debian 12 64-bit bot, but on 32-bit.

I got a couple of build errors, which this commit fixes.

* Source/JavaScriptCore/CMakeLists.txt:
* Source/WebCore/dom/EventTarget.h:

Canonical link: https://commits.webkit.org/300313@main
